src/components/app/KeyboardShortcut.js

```diff
@@ -0,0 +1,270 @@
+            id="AppKeyboardShortcutsHeaderTitle"
+          >
+            Keyboard shortcuts
+          </div>
+          <button
+            type="button"
+            className="appKeyboardShortcutsHeaderClose"
+            onClick={this._handleCloseClick}
+          >
+            Close
+          </button>
+        </div>
+        <div className="appKeyboardShortcutsContent">
+          <div className="appKeyboardShortcutsColumn">
+            <h2>Call Tree</h2>
+            <Shortcut label="Close call node" shortcut="ArrowLeft" />
+            <Shortcut label="Open call node" shortcut="ArrowRight" />
+            <Shortcut label="Open all child call nodes" shortcut="*" />
+            <Shortcut
+              label="Copy call node label"
+              shortcut="ctrl,c"
+              macShortcut="cmd,c"
+            />
+
+            <h2>Flame Graph</h2>
+            <Shortcut label="Move view up" shortcut="w" />
+            <Shortcut label="Move view down" shortcut="s" />
+            <Shortcut label="Move selection up" shortcut="ArrowUp" />
+            <Shortcut label="Move selection down" shortcut="ArrowDown" />
+            <Shortcut label="Move selection left" shortcut="ArrowLeft" />
+            <Shortcut label="Move selection right" shortcut="ArrowRight" />
+
+            <h2>Marker Table</h2>
+            <Shortcut
+              label="Copy marker label"
+              shortcut="ctrl,c"
+              macShortcut="cmd,c"
+            />
+          </div>
+          <div className="appKeyboardShortcutsColumn">
+            <h2>Timeline</h2>
+            <Shortcut
+              label="Select multiple threads"
+              shortcut="ctrl,click"
+              macShortcut="cmd,click"
+            />
+
+            <h2>Call Tree Transforms</h2>
+            <Shortcut label="Merge function" shortcut="m" />
+            <Shortcut label="Merge node only" shortcut="M" />
+            <Shortcut label="Focus on function" shortcut="f" />
+            <Shortcut label="Focus on subtree only" shortcut="F" />
+            <Shortcut label="Collapse function" shortcut="c" />
+            <Shortcut label="Collapse library" shortcut="C" />
+            <Shortcut label="Drop samples with this function" shortcut="d" />
+          </div>
+        </div>
+      </>
+    );
+  }
+
+  render() {
+    const { wrapperClassName, children } = this.props;
+    const { isOpen } = this.state;
+    return (
+      <>
+        <div
+          className={classNames(wrapperClassName, {
+            appKeyboardShortcutsNoInteract: isOpen,
+          })}
+        >
+          {children}
+        </div>
+        <div
+          className={classNames({ appKeyboardShortcuts: true, open: isOpen })}
+        >
+          <div
+            className="appKeyboardShortcutsBox"
+            ref={this._focusArea}
+            tabIndex="0"
+            role="dialog"
+            aria-modal="true"
+            aria-labelledby="AppKeyboardShortcutsHeaderTitle"
+          >
```

Comment:
Is there a reason why we always render this div even when it's closed?

Reply:
Yes, it's so that we can always have the `_focusArea` ref available, since we need to manually manage the focus outside of the React update cycles.

Reply:
```
        {/* Always render this div so that we can target the _focusArea ref outside
            of the React life-cycle. The keyboard shortcuts will only render if the
            modal is actual open. */}
```

I added a clarifying comment.